core/src/main/scala/cats/instances/vector.scala

```diff
@@ -82,6 +82,12 @@ trait VectorInstances extends cats.kernel.instances.VectorInstances {
           G.map2Eval(f(a), lgvb)(_ +: _)
         }.value
 
+      override def mapWithIndex[A, B](fa: Vector[A])(f: (A, Int) => B): Vector[B] =
+        fa.view.zipWithIndex.map(ai => f(ai._1, ai._2)).toVector
```

Comment:
can we benchmark with `fa.iterator.zipWithIndex.map(...`, I think `.iterator` may be faster than `.view`